### PR TITLE
:bug: Read total body while error hanppend on binary protocol

### LINKED ISSUE #74: Binary protocol bug
In cases when `ResponseStatus::KeyNotFound`, `ResponseStatus::KeyExists` is returned by the server, there can be a string body after the header denoting the error. Currently we don't read the body and this causes `ClientError`s when using the command after these statuses are returned.

Example:

```
use memcache::Client;
let mut client = Client::connect("memcache://localhost:11211").unwrap();
client.add("key", "value", 0).unwrap();
client.add("key", "value", 0).unwrap();
client.add("key", "value", 0).unwrap();
```

The third `add` will panic here. I think this is also applicable for `append`, `replace` etc. also.

The bug is [here](https://github.com/aisk/rust-memcache/blob/66bd9671df31377f3edf9e4fce7d7aa11615d23f/src/protocol/binary_packet.rs#L106) It should check if  `value_length` in `total_body_length` is `0`.

### PR DESCRIPTION
**Pull request by aisk**
close #74

### CODE REVIEW

**Comment by aisk** [comment]
Hi @letmutx do you have interests on review this?

File: src/protocol/binary_packet.rs [hunk 1]
```diff
@@ -121,6 +121,8 @@ pub fn parse_cas_response<R: io::Read>(stream: &mut R) -> Result<bool, MemcacheE
 pub fn parse_header_only_response<R: io::Read>(reader: &mut R) -> Result<(), MemcacheError> {
     let header = PacketHeader::read(reader)?;
     if header.vbucket_id_or_status != ResponseStatus::NoError as u16 {
+        let mut buffer = vec![0; header.total_body_length as usize];
```
Comment (letmutx) [inline src/protocol/binary_packet.rs]: Let's move this outside of the `if`? If  `extras_length != 0` and `ResponseStatus == NoError`, we might still leave out bytes on the socket.

**Comment by aisk** [comment]
`parse_simple_response` ?

**Comment by letmutx** [comment]
I guess `parse_response` is sufficient.

**Comment by aisk** [comment]
@letmutx Updated

**Comment by letmutx** [comment]
Looks good to me. Thanks for the quick fix.

**Comment by letmutx** [comment]
Hey @aisk Can you please publish to crates.io so that I can publish my crate.

**Comment by aisk** [comment]
Hi @letmutx do you have any deadlines? I think if you're not in a hurry, we can wait for #75 got fixed and not leave a bug on next release?

**Comment by letmutx** [comment]
#75 is a breaking change. Right now there aren't any breaking changes. I'd say let's release v0.13.2 with the current changes and add #75 in v0.14.0

**Comment by aisk** [comment]
We added some APIs and functions (CAS and TLS), so as semver said, we should bump minor version.

**Pull request merged**